### PR TITLE
Modernize tutorial final code

### PR DESCRIPTION
**Pull request by rjrjr**
_There are a lot of indentation changes here, good idea to hide whitespace while reviewing._

It's time to get the Tutorial caught up with undeprecated API. This PR modifies only the final tutorial module. Once we're happy with the code, I'll back port the changes and update the prose in follow ups.

- Use `AndroidScreen` and drop `ViewRegistry`
- Use `View.setBackHandler`
- Use `TextController`
- More consistent naming, code style for actions and event handlers in `Screen` renderings

I did not introduce `RenderContext.eventHandler {}`, that seems like it would just be confusing to a newcomer.

Also not introducing big new blocks of material, in particular not introducing `Overlay`. I do think we should do that, but for this release I just want to focus on getting the deprecated code deleted.

### CODE REVIEW

File: samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoListWorkflow.kt [hunk 1]
```diff
@@ -29,14 +29,15 @@ object TodoListWorkflow : StatelessWorkflow<ListProps, Output, TodoListScreen>()
...
+      username = renderProps.username,
+      todoTitles = titles.map { it.textValue },
+      onTodoSelected = { context.actionSink.send(selectTodo(it)) },
+      onBackClick = { context.actionSink.send(postGoBack) },
+      onAddClick = { context.actionSink.send(postNewTodo) }
     )
   }
 
-  private fun onBack() = action {
+  private val postGoBack = action {
```
Comment (steve-the-edwards) [inline samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoListWorkflow.kt:40]: same comments.

File: samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoModel.kt [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
-  val note: String
-)
+  val title: TextController,
+  val note: TextController
+) {
+  constructor(
+    title: String,
+    note: String
+  ) : this(TextController(title), TextController(note))
+}
```
Comment (steve-the-edwards) [inline samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoModel.kt:15]: is this used directly in the rendering?

File: workflow-ui/container-common/src/main/java/com/squareup/workflow1/ui/modal/AlertContainerScreen.kt [hunk 3]
```diff
@@ -16,10 +16,10 @@ import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
...
 // @Deprecated(
-//   "Use BodyAndModalsScreen and AlertOverlay",
+//   "Use BodyAndOverlaysScreen and AlertOverlay",
 //   ReplaceWith(
-//     "BodyAndModalsScreen<B>(beneathModals, modals)",
-//     "com.squareup.workflow1.ui.container.BodyAndModalsScreen"
+//     "BodyAndOverlaysScreen<B>(beneathModals, modals)",
+//     "com.squareup.workflow1.ui.container.BodyAndOverlaysScreen"
 //   )
 // )
```
Comment (steve-the-edwards) [inline workflow-ui/container-common/src/main/java/com/squareup/workflow1/ui/modal/AlertContainerScreen.kt:24]: will need rebase I take it?

**Review by steve-the-edwards — APPROVED** [review]

File: samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoEditWorkflow.kt [hunk 4]
```diff
@@ -54,33 +56,20 @@ object TodoEditWorkflow : StatefulWorkflow<EditProps, State, Output, TodoEditScr
...
-  internal fun onTitleChanged(title: String) = action {
-    state = state.withTitle(title)
-  }
-
-  internal fun onNoteChanged(note: String) = action {
-    state = state.withNote(note)
-  }
-
-  private fun onDiscard() = action {
+  private val postDiscard = action {
```
[the remark this answers is not in the document]
Comment (rjrjr) [reply samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoEditWorkflow.kt:66]: > LIke is the action about what it does (posting a save) or what it responds to (on "Save" clicked)? I feel like we've often used the latter, so maybe that's what is tripping me up. Maybe the former is preferrable?

I like the `*Action` notion, will adopt.

File: samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoListScreen.kt [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
  * `val onButtonTapped: () -> Unit`.
  */
+@OptIn(WorkflowUiExperimentalApi::class)
 data class TodoListScreen(
   val username: String,
   val todoTitles: List<String>,
   val onTodoSelected: (Int) -> Unit,
-  val onBack: () -> Unit
-)
+  val onBackClick: () -> Unit,
```
[the remark this answers is not in the document]
Comment (rjrjr) [reply samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoListScreen.kt:19]: 🤔

File: samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoModel.kt (hunk 2 above)
Comment (rjrjr) [reply samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoModel.kt:15]: Yes. That's the normal pattern for `TextController`, which is why I've pushed back when anyone talks about generalizing it.

File: workflow-ui/container-common/src/main/java/com/squareup/workflow1/ui/modal/AlertContainerScreen.kt (hunk 3 above)
Comment (rjrjr) [reply workflow-ui/container-common/src/main/java/com/squareup/workflow1/ui/modal/AlertContainerScreen.kt:24]: Yup!

File: samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoEditWorkflow.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (rjrjr) [reply samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoEditWorkflow.kt:66]: Actually…

What I like about the awkward `post*` names was that they communicated what was going to happen in response to the UI event. I don't think that's an implementation detail, it helps make visible what the role of this workflow is in the overall structure. And once I've put the names in that very active format, the `*Action` suffix starts to sound noisy. 

WDYT:

```kotlin
      onTodoSelected = { context.actionSink.send(requestSelection(it)) },
      onBackPressed = { context.actionSink.send(requestGoBack) },
      onAddPressed = { context.actionSink.send(requestNew) }
```

**Pull request closed without merging**